providers/statsig/src/main/java/dev/openfeature/contrib/providers/statsig/StatsigProvider.java

```diff
@@ -0,0 +1,320 @@
+        mutableContext.add("value", Structure.mapToStructure(layer.getValue()));
+        mutableContext.add("ruleID", layer.getRuleID());
+        mutableContext.add("groupName", layer.getGroupName());
+        List<Value> secondaryExposures = new ArrayList<>();
+        layer.getSecondaryExposures().forEach(secondaryExposure -> {
+                Value value = Value.objectToValue(secondaryExposure);
+                secondaryExposures.add(value);
+            }
+        );
+        mutableContext.add("secondaryExposures", secondaryExposures);
+        mutableContext.add("allocatedExperiment", layer.getAllocatedExperiment());
+        return new Value(mutableContext);
+    }
+
+    @NotNull
+    private static FeatureConfig parseFeatureConfig(EvaluationContext ctx) {
+        Value featureConfigValue = ctx.getValue(FEATURE_CONFIG_KEY);
+        if (featureConfigValue == null) {
+            throw new IllegalArgumentException("feature config not found at evaluation context.");
+        }
+        if (!featureConfigValue.isStructure()) {
+            throw new IllegalArgumentException("feature config is not a structure.");
+        }
+        Structure featureConfigStructure = featureConfigValue.asStructure();
+        Value typeValue = featureConfigStructure.getValue("type");
+        if (typeValue == null) {
+            throw new IllegalArgumentException("feature config type is missing");
+        }
+        FeatureConfig.Type type = FeatureConfig.Type.valueOf(typeValue.asString());
+        Value nameValue = featureConfigStructure.getValue("name");
+        if (nameValue == null) {
+            throw new IllegalArgumentException("feature config name is missing");
+        }
+        String name = nameValue.asString();
+        return new FeatureConfig(type, name);
+    }
+
+    private void verifyEvaluation() throws ProviderNotReadyError, GeneralError {
+        if (!ProviderState.READY.equals(state)) {
+
+            /*
+            According to spec Requirement 2.4.5:
+            "The provider SHOULD indicate an error if flag resolution is attempted before the provider is ready."
+            https://github.com/open-feature/spec/blob/main/specification/sections/02-providers.md#requirement-245
+             */
+            if (ProviderState.NOT_READY.equals(state)) {
+                throw new ProviderNotReadyError(PROVIDER_NOT_YET_INITIALIZED);
+            }
+            throw new GeneralError(UNKNOWN_ERROR);
+        }
+    }
+
+    @SneakyThrows
+    @Override
+    public void shutdown() {
+        log.info("shutdown");
+        Statsig.shutdown();
+        state = ProviderState.NOT_READY;
+    }
+
+    /**
+     * Feature config, as required for evaluation.
+     */
+    @AllArgsConstructor
+    @Getter
+    public static class FeatureConfig {
+
+        /**
+         * Type.
+         *  CONFIG: Dynamic Config
+         *  LAYER: Layer
+         */
+        public enum Type {
+            CONFIG, LAYER
```

Comment:
Nit: in statsig , we have concept of experiment, but down the line, it's essential same as dynamicConfig. So consider add experiment

Reply:
According to [Statsig docs](https://docs.statsig.com/server/javaSdk#getting-an-layerexperiment):
> We offer two APIs, but we recommend the use of [layers](https://docs.statsig.com/layers) to enable quicker iterations with parameter reuse.

Then I thought to support here only the recommended use.